src/flint/types/fq_default.pyx

```diff
@@ -0,0 +1,790 @@
+        """"
+        Return the multiplicative order of the finite field
+
+            >>> gf = fq_default_ctx(5, 2)
+            >>> gf.multiplicative_order()
+            24
+        """
+        return self.order() - 1
+
+    def modulus(self):
+        """
+        Return the modulus from the context as an fmpz_mod_poly type
+
+            >>> gf = fq_default_ctx(5, 2)
+            >>> gf.modulus()
+            x^2 + 4*x + 2
+
+        """
+        cdef fmpz_mod_poly_ctx ctx
+        cdef fmpz_mod_poly pol
+        ctx = fmpz_mod_poly_ctx(self.prime())
+        pol = ctx.new_ctype_poly()
+        fq_default_ctx_modulus(pol.val, self.val)
+        return pol
+
+    def zero(self):
+        """
+        Return the zero element
+
+            >>> gf = fq_default_ctx(5)
+            >>> gf.zero()
+            0
+        """
+        cdef fq_default res
+        res = self.new_ctype_fq_default()
+        res.ctx = self
+        fq_default_zero(res.val, self.val)
+        return res
+
+    def one(self):
+        """
+        Return the unit element
+
+            >>> gf = fq_default_ctx(5)
+            >>> gf.one()
+            1
+        """
+        cdef fq_default res
+        res = self.new_ctype_fq_default()
+        res.ctx = self
+        fq_default_one(res.val, self.val)
+        return res
+
+    def gen(self):
+        """
+        Return the one element
+
+            >>> gf = fq_default_ctx(5, 2, var="w")
+            >>> gf.gen()
+            w
+        """
+        cdef fq_default res
+        res = self.new_ctype_fq_default()
+        res.ctx = self
+        fq_default_gen(res.val, self.val)
+        return res
+
+    def random_element(self, not_zero=False):
+        r"""
+        Return a random element of the finite field
+
+            >>> gf = fq_default_ctx(163, 3)
+            >>> a = gf.random_element()
+            >>> type(a) is fq_default
+            True
+            >>> a = gf.random_element(not_zero=True)
+            >>> not a.is_zero()
+            True
+        """
+        cdef fq_default res
+        res = self.new_ctype_fq_default()
+        res.ctx = self
+        if not_zero:
+            fq_default_rand_not_zero(res.val, global_random_state, self.val)
+        else:
+            fq_default_rand(res.val, global_random_state, self.val)
+        return res
+
+    cdef new_ctype_fq_default(self):
+        return fq_default.__new__(fq_default, None, self)
+
+    cdef set_list_as_fq_default(self, fq_default_t fq_ele, obj):
+        cdef fmpz_poly poly
+        poly = fmpz_poly.__new__(fmpz_poly)
+        fmpz_poly_set_list(poly.val, obj)
+
+        # Now set the value from the fmpz_poly
+        fq_default_set_fmpz_poly(fq_ele, poly.val, self.val)
+
+        return 0
+
+    cdef set_any_as_fq_default(self, fq_default_t fq_ele, obj):
+        # Converts the list to an fmpz_poly and then sets from this
+        if typecheck(obj, list):
+            return self.set_list_as_fq_default(fq_ele, obj)
```

Comment:
I think it is a bit dangerous to accept a list here if this function is used as part of `__add__` etc.

Reply:
It is even used in `__eq__`:
```python
In [4]: ctx = flint.fq_default_ctx(5, 2, 'y')

In [5]: y = ctx.gen()

In [6]: y
Out[6]: y

In [7]: y == [0, 1]
Out[7]: True
```

Reply:
I'd like to be able to do `GF([1,2,3])` though, as elements of the extensions are essentially polynomials

Reply:
but maybe we need a different call for `__call__` on the context and on the arithmetic type casting

Reply:
I think that `__call__` is definitely a different case. Implicit coercions need to be limited especially if they are used in things like `__eq__`.

When someone calls `GF(obj)` that is an explicit conversion which is a very different situation.

Reply:
OK this has been fixed:

```py
>>> from flint import fq_default_ctx
>>> gf = fq_default_ctx(5)
>>> gf(0) == [0]
False
>>> gf.gen() == [0, 1]
False
```